tests/server.features/steps/langserver_steps.py

```diff
@@ -45,19 +56,100 @@ def step_impl(context):
         'method': 'textDocument/didSave',
         'params': {
             'textDocument': {
-                'uri': 'file:///Users/mock-user/non-exist.py'
-            }
+                'uri': 'file:///Users/mock-user/non-exist.py',
+            },
+        },
+        'jsonrpc': '2.0',
+    }
+    context.langServer._endpoint.consume(request)
+    context._diagCount = 0
+
+
+@when('I send a did_save request about a existing file to the server')
+def step_impl(context):
+    thisfile = os.path.realpath(__file__)
+    thisdir = os.path.dirname(thisfile)
+    parturi = os.path.join(thisdir, '../../resources', 'unqualified.py')
+    absparturi = os.path.abspath(parturi)
+
+    request = {
+        'method': 'textDocument/didSave',
+        'params': {
+            'textDocument': {
+                'uri': 'file://{}'.format(absparturi),
+            },
+        },
+        'jsonrpc': '2.0',
+    }
+    context.langServer._endpoint.consume(request)
+    context._diagCount = 4
+
+
+@when('I send a did_save request on a file with no coafile to server')
+def step_impl(context):
+    somefile = tempfile.NamedTemporaryFile(delete=False)
+    somefilename = somefile.name
+    somefile.close()
+
+    request = {
+        'method': 'textDocument/didSave',
+        'params': {
+            'textDocument': {
+                'uri': 'file://{}'.format(somefilename),
+            },
         },
-        'jsonrpc': '2.0'
+        'jsonrpc': '2.0',
     }
-    context.langServer.handle(None, request)
+    context.langServer._endpoint.consume(request)
+    context._diagCount = 0
 
 
-@then('it should send a publishDiagnostics request')
+@then('it should send a publishDiagnostics response')
```

Comment:
this is a separate bug?

create a newcomer issue for this change.  get other developers involved ;-)